### DISCUSSION TITLE
Update tabs count from filters

### DISCUSSION THREAD

**Discussion by sasaorasanin** [filamentphp/filament#10803] [Help] [answered]
Is it possible to  catch table query after applying filters and use it in tabs badge?
```php
->badge($query->count())
```

Something like `$this->getTableQuery() ||  $this->getFilteredTableQuery()` but `getFilteredTableQuery` returns cannot be accessed before init and `getTableQuery` doesn't have filters/tab filters..

On `$this->getTable()->getQuery()` I get nginx timeout error.

**Comment by leandrocfe** [comment]
```php
->badge(fn () => $this->getTableRecords()->count())
```

**Reply by sasaorasanin** [reply]
Because of pagination, it limits it to max per page, where I would need total with additional conditions..

**Reply by sasaorasanin** [reply, marked as answer]
but yeah, this works: `->badge(fn () => $this->getFilteredTableQuery()->clone()->where('is_completed', true)->count())`

**Comment by barocode** [comment]
Filament 4.x found this solution: 

```->badge(fn () => $this->applyFiltersToTableQuery($this->getModel()::query())->where('is_completed', true)->count())```

It's apply to query only filters and when you change tab with modified query using like: 

```->modifyQueryUsing(fn (Builder $query) => $query->where('is_completed', true))``` 

then you still has correct counts on other tabs.
